google-cloud-storage/lib/google/cloud/storage/project.rb

```diff
@@ -384,6 +385,91 @@ def create_bucket bucket_name, acl: nil, default_acl: nil,
           Bucket.from_gapi gapi, service, user_project: user_project
         end
 
+        ##
+        # Creates a new HMAC key.
+        #
+        # @param [String] service_account_email The email address of the service
+        #   account. Used to create the HMAC key.
+        # @param [String] project_id The project ID associated with
+        #   `service_account_email`, if `service_account_email` belongs to a
+        #   project other than the currently authenticated project. Optional. If
+        #   not provided, the project ID for the current project will be used.
+        # @param [String] user_project If this parameter is set to a project ID
+        #   other than the current project, and that project is authorized for
+        #   the currently authenticated service account, transit costs will be
+        #   billed to the given project. The default is `nil`.
+        #
+        # @return [Google::Cloud::Storage::HmacKey]
+        #
+        def create_hmac_key service_account_email, project_id: nil,
+                            user_project: nil
+          gapi = service.create_hmac_key service_account_email,
+                                         project_id: project_id,
+                                         user_project: user_project
+          HmacKey.from_gapi gapi, service, user_project: user_project
```

Comment:
Store `project_id` after create_hmac_key.

Reply:
The `gapi` variable here holds the `Google::Apis::StorageV1::HmacKeyMetadata` instance returned by the RPCs (including `service.get_hmac_key`) and will be held in the `HmacKey` instance, so that's OK isn't it?